diff --git a/build/lib/release.sh b/build/lib/release.sh
index 4a086bdb39d..1661db97334 100644
--- a/build/lib/release.sh
+++ b/build/lib/release.sh
@@ -337,6 +337,8 @@ function kube::release::create_docker_images_for_server() {
         rm -rf "${docker_build_path}"
         mkdir -p "${docker_build_path}"
         ln "${binary_file_path}" "${docker_build_path}/${binary_name}"
+        ln "${binary_dir}/kubectl" "${docker_build_path}/kubectl"
+        ln LICENSE "${docker_build_path}/LICENSE"
 
         local build_log="${docker_build_path}/build.log"
         if ! DOCKER_CLI_EXPERIMENTAL=enabled "${DOCKER[@]}" buildx build \
diff --git a/build/server-image/Dockerfile b/build/server-image/Dockerfile
index 09d2cd1e691..92a633e273d 100644
--- a/build/server-image/Dockerfile
+++ b/build/server-image/Dockerfile
@@ -17,6 +17,19 @@
 ARG BASEIMAGE
 ARG BINARY
 
+FROM --platform=linux/$BUILDARCH busybox:latest AS directories
+RUN mkdir -p /tmp/rootfs/var/run/kubernetes /tmp/rootfs/tmp /tmp/rootfs/etc/ssl/certs /tmp/rootfs/etc/ca-certificates /tmp/rootfs/etc/kubernetes/pki /tmp/rootfs/usr/local/share/ca-certificates /tmp/rootfs/usr/share/ca-certificates
+RUN chmod -R 775 /tmp/rootfs/var/run/kubernetes /tmp/rootfs/tmp /tmp/rootfs/etc/ssl/certs /tmp/rootfs/etc/ca-certificates /tmp/rootfs/etc/kubernetes /tmp/rootfs/usr/local/share/ca-certificates /tmp/rootfs/usr/share/ca-certificates
 
 FROM "${BASEIMAGE}"
+LABEL name="HariKube"
+LABEL vendor="inspirNation Bt."
+LABEL version="1.35.6"
+LABEL release="0"
+LABEL summary="Kubernetes HariKube Edition"
+LABEL description="Kubernetes HariKube edition is a horizontally scalable Kubernetes edition."
+LABEL maintainer="richard.kovacs@harikube.com"
+COPY LICENSE /licenses/LICENSE
+COPY --from=directories --chown=65534:0 /tmp/rootfs/* /
+USER 0
 COPY --chmod=755 ${BINARY} /usr/local/bin/${BINARY}
diff --git a/build/server-image/kube-apiserver/Dockerfile b/build/server-image/kube-apiserver/Dockerfile
index 27129d29bd9..df114656cb9 100644
--- a/build/server-image/kube-apiserver/Dockerfile
+++ b/build/server-image/kube-apiserver/Dockerfile
@@ -25,6 +25,21 @@ COPY --chmod=755 ${BINARY} /${BINARY}
 # non-root and still listen on port less than 1024
 RUN setcap cap_net_bind_service=+ep /${BINARY}
 
+FROM --platform=linux/$BUILDARCH busybox:latest AS directories
+RUN mkdir -p /tmp/rootfs/var/run/kubernetes /tmp/rootfs/tmp /tmp/rootfs/etc/ssl/certs /tmp/rootfs/etc/ca-certificates /tmp/rootfs/etc/kubernetes/pki /tmp/rootfs/usr/local/share/ca-certificates /tmp/rootfs/usr/share/ca-certificates
+RUN chmod -R 775 /tmp/rootfs/var/run/kubernetes /tmp/rootfs/tmp /tmp/rootfs/etc/ssl/certs /tmp/rootfs/etc/ca-certificates /tmp/rootfs/etc/kubernetes /tmp/rootfs/usr/local/share/ca-certificates /tmp/rootfs/usr/share/ca-certificates
+
 FROM --platform=linux/$TARGETARCH ${BASEIMAGE}
+LABEL name="HariKube"
+LABEL vendor="inspirNation Bt."
+LABEL version="1.35.6"
+LABEL release="0"
+LABEL summary="Kubernetes HariKube Edition"
+LABEL description="Kubernetes HariKube edition is a horizontally scalable Kubernetes edition."
+LABEL maintainer="richard.kovacs@harikube.com"
+COPY LICENSE /licenses/LICENSE
+COPY --from=directories --chown=65534:0 /tmp/rootfs/* /
+USER 0
 ARG BINARY
 COPY --from=0 /${BINARY} /usr/local/bin/${BINARY}
+COPY --chmod=755 kubectl /usr/local/bin/kubectl
diff --git a/hack/lib/version.sh b/hack/lib/version.sh
index 65595e63fe9..c1a88b32e58 100644
--- a/hack/lib/version.sh
+++ b/hack/lib/version.sh
@@ -91,7 +91,7 @@ kube::version::get_version_vars() {
         # git describe --dirty only considers changes to existing files, but
         # that is problematic since new untracked .go files affect the build,
         # so use our idea of "dirty" from git status instead.
-        KUBE_GIT_VERSION+="-dirty"
+        KUBE_GIT_VERSION+=""
       fi
 
 
